File: Problems/1305-number-of-visible-people-in-a-queue/number-of-visible-people-in-a-queue.py
class Solution:
    def canSeePersonsCount(self, heights: List[int]) -> List[int]:
        stack = []
        results = []
        for height in reversed(heights):
            items_popped = 0
            while stack and stack[-1] <= height:
                stack.pop()
                items_popped += 1
            results.append(items_popped + (1 if stack else 0))
            stack.append(height)

        return list(reversed(results))

File: Problems/1305-number-of-visible-people-in-a-queue/number-of-visible-people-in-a-queue.cpp
typedef long long ll;
class Solution {
public:
    vector<int> canSeePersonsCount(vector<int>& heights) {
        vector<int>ans;
        stack<ll>s;
        
        int cnt=0;
        for(int i=heights.size()-1;~i;i--)
        {
            cnt = 0;
            while(!s.empty() && s.top()<heights[i]){
                cnt++;
                s.pop();
            }
            if(!s.empty())cnt++;
            ans.push_back(cnt);
            s.push(heights[i]);
        }
        reverse(ans.begin(),ans.end());
        return ans;
    }
};